theano/scalar/basic_scipy.py

```diff
@@ -388,8 +417,13 @@ def impl(self, x):
         else:
             super(J1, self).impl(x)
 
-    def grad(self, inp, grads):
-        raise NotImplementedError()
+    def grad(self, inputs, grads):
+        if imported_scipy_special:
+            x, = inputs
+            gz, = grads
+            return [gz * (j0(x) - jv(2, x)) / 2.]
+        else:
+            super(J1, self).grad(v, x)
```

Comment:
The other grad() method in this file don't use "if imported_scipy_special". Why did you use it? If there isn't a specific reason, remove the if. The other op already do the check.

This is for all the ops.